File: algorithms/js/binarySearch.js
function binarySearch(arr, target) {
  if (arr.length == 0) {
    return false;
  }

  let mid = Math.floor(arr.length / 2);
  let left = arr.slice(0, mid);
  let right = arr.slice(mid + 1);

  if (arr[mid] > target) {
    return binarySearch(left, target);
  } else if (arr[mid] < target) {
    return binarySearch(right, target);
  } else {
    return true;
  }
}

console.log(binarySearch([5, 10, 12, 15, 20, 30, 70], 12)); // => true
console.log(binarySearch([5, 10, 12, 15, 20, 30, 70], 24)); // => false

File: algorithms/ruby/binary_search.rb
def binary_search(arr, target)
  return false if arr.length == 0

  mid = arr.length / 2
  left = arr.take(mid)
  right = arr.drop(mid + 1)

  if arr[mid] > target
    return binary_search(right, target)
  elsif arr[mid] < target
    subresult = binary_search(right, target)
    return subresult == false ? false : subresult + mid + 1
  else
    return true
  end
end

p binary_search([5, 10, 12, 15, 20, 30, 70], 12) == true
p binary_search([5, 10, 12, 15, 20, 30, 70], 24) == false